clients/IntermediaryClient.ts

```diff
@@ -58,18 +59,24 @@ export class IntermediaryCoordinator {
       // todo: return a failure message to the sender?
     }
 
+    const targetNetwork = await targetClient.getHostNetwork();
+
+    if (targetNetwork !== htlc.invoice.chain) {
+      htlc.invoice = convertInvoice(htlc.invoice, targetNetwork);
+    }
+
     const fee = 0; // for example
+
     const updatedState = await targetClient.addHTLC(
-      htlc.amount - BigInt(fee),
-      htlc.hashLock,
+      htlc.invoice.amount - BigInt(fee),
```

Comment:
Does the fee amount get routed to the Intermediary somehow? If so, how does that happen?

Reply:
If a non-zero fee is deducted here, then when all of the payments unlock Irene will have had a net gain of `fee`.